models/concerns/distance_matrix.rb

```diff
@@ -72,6 +72,8 @@ def compute_need_matrix(&block)
           distance: (router_matrices[dimensions.index(:distance)] if dimensions.index(:distance)),
           value: (router_matrices[dimensions.index(:value)] if dimensions.index(:value))
         )
+        raise RouterError, 'Negative value provided by router' if [m.time, m.distance, m.value].compact.any?{ |matrix| matrix.flatten.any?(&:negative?) }
```

Comment:
```suggestion
        raise RouterError, 'Negative value provided by router' if [m.time, m.distance, m.value].any?{ |matrix| matrix&.any?{ |row| row.any?(&:negative?) } }
```
I did the benchmark during the meeting, it turns out without flatten/compact, the routine is ~2.4x faster if there are no negative values and if there are negative values, it is thousands times (~5K) faster depending on the location of the negative value..